packages/atomic/src/components/common/generated-answer/generated-answer-common.tsx

```diff
@@ -313,18 +322,28 @@ export class GeneratedAnswerCommon {
           </GeneratedContentContainer>
         ) : null}
 
-        {!this.hasRetryableError &&
-          this.isAnswerVisible &&
-          !this.props.getGeneratedAnswerState()?.isStreaming && (
-            <div
-              part="generated-answer-footer"
-              class="flex justify-end mt-6 text-neutral-dark text-xs"
-            >
-              <slot name="disclaimer" slot="disclaimer">
-                {this.props.getBindings().i18n.t('generated-answer-disclaimer')}
-              </slot>
-            </div>
-          )}
+        {!this.hasRetryableError && this.isAnswerVisible && (
+          <div part="generated-answer-footer" class="flex justify-end mt-6">
+            <ShowButton
+              i18n={this.props.getBindings().i18n}
+              onClick={() =>
+                this.props.setIsCollapsed(!this.props.getIsCollapsed())
+              }
+              isStreaming={this.props.getGeneratedAnswerState()?.isStreaming}
+              collapsible={this.props.collapsible}
+              isCollapsed={this.props.getIsCollapsed()}
+            ></ShowButton>
+            {!this.props.getGeneratedAnswerState()?.isStreaming && (
+              <div class="text-neutral-dark text-xs">
+                <slot name="disclaimer" slot="disclaimer">
+                  {this.props
+                    .getBindings()
+                    .i18n.t('generated-answer-disclaimer')}
+                </slot>
+              </div>
+            )}
+          </div>
+        )}
```

Comment:
[ :rotating_light: nitpicking alert :rotating_light: ]

The short circuiting pattern inside a render (`true && ...render`) is making my soft brain hurt, usually.

When I am about to do this, this is usually the signal for me to split the component in more clear an consise units with each their own rendering condition(s).

so, in short I usually prefer this :point_down: to short circuiting. This is lengthier and dumber. But I guess I just like long dumb stuff.

```
const Patate = () => {
    if (!canRenderPatate) {
        return null;
    }
    return <div>Patate</div>;
};

const Pwell = () => {
    if (!canRenderPwell) {
        return null;
    }
    return <div>Pwell</div>;
};

const PatatePwell = () => (
    <div>
        <Patate />
        <Pwell />
    </div>
);
```

Roger out, I should not bother you again with this point of view of mine. If in most case you make sure the component is readable, its a matter of preferences in the end.

Reply:
Yes agreed that this file is getting out of hand. It will not be easily re-usable in other context than classic search and insight panel very well as it stands right now.

ie: If/when we ever need it for commerce, we'll need to refactor this file.

Reply:
Yes agreed, I added this section to match what we have until now, But I will change the whole file to be more readable